google/ggrc-core
# GGRC-3463: FE-Bulk Task Update on My Tasks page

NOTE: 
- [x] This PR should be checked after   #6672 
- [x] This PR should be checked after   #6703 
- [x] This PR should be checked after   #6704 
- [x] This PR should be checked after   #6705 
- [x] This PR should be checked after   #6708 

# Issue description

Add ability to update CycleTaskGroupObjectTask state in bulk.

# Steps to test the changes

1) WHERE?
My Work - Tasks page. 

2) BULK UPDATE in tree view:
'BULK UPDATE' option should be displayed under 3 dots on My Tasks page.
On clicking 'BULK UPDATE', 'Bulk Update' pop-up should be opened. 

3) BULK UPDATE pop-up:
Advanced filters on BULK UPDATE pop-up:
 - Task State (Assigned / In Progress / Finished / Declined / Deprecated / Verified); 
 - Task Attributes (Summary / Assignee / Start Date / Due Date)
 
List of tasks related to logged-in user should be displayed on pop-up (same as on My Tasks page); 
 - Task Title, Task State, Task Assignees, Task Start Date, Task Due Date columns should be displayed by default for tasks list. 
 - Cycle title, Task Last Updated, Task Last Updated By columns should be hidden by default. 
 - Task state filed should be displayed with options: In Progress / Declined / Finished / Verified

4) MESSAGES on Bulk Update pop-up, depending on value selected in 'State' dropdown:
 - In Progress: "Please be aware that Finished, Declined and Verified tasks cannot be moved to In Progress state." 
 - Finished: "Please be aware that Assigned and Verified tasks cannot be moved to Finished state." 
 - Declined: "Please be aware that Assigned, In Progress and Verified tasks cannot be moved to Declined state." 
 - Verified: "Please be aware that Assigned, In Progress and Declined tasks cannot be moved to Verified state." 
 - Deprecated: Notes: no message. 

5) UPDATE button:
 - On clicking UPDATE button, pop-up should be closed. 
 - On clicking 'UPDATE' button, message 'Task update is in progress. This may take several minutes' should be shown while task updating is in progress.' should be shown while task are being updated. 
Message '# tasks were updated successfully.' / '1 task was updated successfully.' should be shown when updating is finished and at least one task is updated. 
 - After tasks are successfully updated, tree view should be reloaded and display the updated results. 
- Message 'No tasks were updated.' should be shown when updating is finished and no task is updated. 

# Solution description
https://docs.google.com/document/d/1xJWcNy1aBXEjSdZREwW-YJBxYwbPFKEiToUul0POJ2M/edit

# Sanity checklist


# PR Review checklist


NOTE:
- It was agreed with BA that user will be able to filter CTGOT by value in 'Status' dropdown in the scope of this ticket on My Tasks page and Global Search modal - this functionality works out of the box when adding possibility to filter Tasks by Status in Bulk Update modal.
- 'Task State' column was made default on Global Search and CTGOT tree views.
